spec/requests/widgets_spec.rb

```diff
@@ -62,43 +65,66 @@
         expect(response).to have_http_status(:not_found)
       end
 
-      it "generate_content of a single Widget" do
-        miq_widget = FactoryBot.create(:miq_widget)
-
-        api_basic_authorize('widget_generate_content')
-
-        expect(MiqTask.count).to eq(0)
-        post(api_widget_url(nil, miq_widget), :params => gen_request(:generate_content))
-
-        expect_single_action_result(:success => true, :message => /#{miq_widget.id}.* content generation/i, :href => api_widget_url(nil, miq_widget))
-        expect(MiqTask.first.message).to match(/content generation\]\ being run for user/)
+      context "generate_content for a group" do
+        before do
+          stub_settings(::Settings.to_hash.merge(:product => {:report_sync => true}))
+          MiqReport.seed_report("Vendor and Guest OS")
+          feature1 = MiqProductFeature.find_all_by_identifier("dashboard_admin")
```

Comment:
we can share some  code here.
1. We can use `let`'s for lines 72-75 and 101-104.
2. those two `before do` blocks looks same, is it possible to move them under contexts or move them on place `somehow` ?